File: AdvancedMap.Objective-C/AdvancedMap/OfflineMapController.m
//
//  MapPackageController.m
//  AdvancedMap.Objective-C
//
//  Created by Aare Undo on 21/08/2017.
//  Copyright © 2017 Nutiteq. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <CartoMobileSDK/CartoMobileSDK.h>
#import "PackageDownloadBaseController.h"
#import "Sources.h"

@interface OfflineMapController : PackageDownloadBaseController

@end

@implementation OfflineMapController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.contentView = [[PackageDownloadBaseView alloc] init];
    self.view = self.contentView;
    
    NSString *source = CARTO_VECTOR_SOURCE;
    NSString *folder = [self createFolder:@"com.carto.mappackages"];
    
    [self.contentView setManager:source folder:folder];
    [self.contentView setOfflineLayer];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    if (![self.contentView hasLocalPackages]) {
        [self showInformationBanner: @"Click the globe icon and download a map package to continue"];
    }
}

- (void)showInformationBanner: (NSString *)text {
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.contentView.banner showInformationWithText:text autoclose:YES];
    });
}

@end

File: AdvancedMap.Swift/Feature Demo/OfflineMapController.swift
//
//  CountryDownloadController.swift
//  AdvancedMap.Swift
//
//  Created by Aare Undo on 27/06/2017.
//  Copyright © 2017 CARTO. All rights reserved.
//

import Foundation
import UIKit
import CartoMobileSDK

class OfflineMapController : BasePackageDownloadController {
    
    var mapPackageListener: MapPackageListener!
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        contentView = OfflineMapView()
        view = contentView
        
        let folder = Utils.createDirectory(name: "countrypackages")
        contentView.manager = NTCartoPackageManager(source: Routing.MAP_SOURCE, dataFolder: folder)
        
        // Online mode by default
        
        setOnlineMode()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        contentView.addRecognizers()
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        
        contentView.removeRecognizers()
    }
    
    override func downloadComplete(sender: PackageListener, id: String) {
        contentView.downloadComplete(id: id)
        goOffline()
    }
    
    override func setOnlineMode() {
        super.setOnlineMode()
        contentView.setOnlineMode()
    }
    
    override func setOfflineMode() {
        super.setOfflineMode()
        contentView.setOfflineMode()
    }
    
}







